repos/system_upgrade/common/actors/repositoriesmapping/libraries/repositoriesmapping.py

```diff
@@ -1,12 +1,111 @@
+from collections import defaultdict
+import json
+import os
+
 from leapp.exceptions import StopActorExecutionError
-from leapp.libraries.common import config
+from leapp.libraries.common.config.version import get_target_major_version, get_source_major_version
 from leapp.libraries.common.fetch import read_or_fetch
 from leapp.libraries.stdlib import api
-from leapp.models import RepositoriesMap, RepositoryMap
+from leapp.models import RepositoriesMapping, PESIDRepositoryEntry, RepoMapEntry
 from leapp.models.fields import ModelViolationError
 
-REPOMAP_FILE = 'repomap.csv'
-"""Path to the repository mapping file."""
+OLD_REPOMAP_FILE = 'repomap.csv'
+"""The name of the old repository mapping file."""
+
+REPOMAP_FILE = 'repomap.json'
+"""The name of the new repository mapping file."""
+
+
+class RepoMapData(object):
+    VERSION_FORMAT = '1.0.0'
+
+    def __init__(self):
+        self.repositories = []
+        self.mapping = {}
+
+    def add_repository(self, data, pesid):
+        self.repositories.append(PESIDRepositoryEntry(
+            repoid=data['repoid'],
+            channel=data['channel'],
+            rhui=data.get('rhui', ''),
+            repo_type=data['repo_type'],
+            arch=data['arch'],
+            major_version=data['major_version'],
+            pesid=pesid
+        ))
+
+    def get_repositories(self, valid_major_versions):
+        """
+        Return the list of PESIDRepositoryEntry object mathing the specified major versions.
+        """
+        return [repo for repo in self.repositories if repo.major_version in valid_major_versions]
+
+    def add_map(self, src_major_version, dst_major_version, src_pesid, dst_pesid):
+        # NOTE: it could be more simple, but I prefer to be sure the input data
+        # contains just one map per source PESID.
+        key = '{}:{}'.format(src_major_version, dst_major_version)
+        rmap = self.mapping.get(key, defaultdict(set))
+        self.mapping[key] = rmap
+        if isinstance(dst_pesid, list):
+            rmap[src_pesid].update(dst_pesid)
+        else:
+            rmap[src_pesid].add(dst_pesid)
+
+    def get_maps(self, src_major_version, dst_major_version):
+        """
+        Return the list of RepoMapEntry objects for the specified upgrade path.
+
+        IOW, the whole mapping for specified IPU.
+        """
+        key = '{}:{}'.format(src_major_version, dst_major_version)
+        rmap = self.mapping.get(key, None)
+        if not rmap:
+            return None
+        map_list = []
+        for src_pesid in sorted(rmap.keys()):
+            map_list.append(RepoMapEntry(source=src_pesid, target=sorted(rmap[src_pesid])))
+        return map_list
+
+    @staticmethod
+    def deserialize(data):
```

Comment:
```suggestion
    def load_from_dict(data):
```

If I'm reading the code correctly `data` at this point has already been deserialized from JSON.

Reply:
Yes, you are right. the `load*` seems better in this case unless we put the `json.load` inside - but frankly working with deserialized data is simpler for testing, so I prefer the change of the name instead. Same for the other method.

Reply:
The proposed naming change has been implemented :grin:

Reply:
done